R/aggregate_alignment_loanbook_exposure.R

```diff
@@ -85,81 +166,72 @@ aggregate_alignment_loanbook_exposure <- function(data,
-  # high carbon and low carbon technologies, add an empty entry for the other
-  # direction to ensure the aggregation is correct
-  if (level == "bo_po") {
-    aggregate_exposure_company <- aggregate_exposure_company %>%
-      dplyr::mutate(
-        n_directions = dplyr::n(),
-        .by = dplyr::all_of(
-          c(
-            group_vars[!group_vars == "direction"], "name_abcd", "sector",
-            "activity_unit", "loan_size_outstanding_currency"
-          )
-        )
-      )
-
-    single_direction <- aggregate_exposure_company %>%
-      dplyr::filter(
-        .data$n_directions == 1,
-        .data$direction %in% c("buildout", "phaseout"),
-        .data$sector %in% c("automotive", "hdv", "power")
-      )
-
-    opposite_direction <- single_direction %>%
-      dplyr::mutate(
-        direction = dplyr::if_else(
-          .data$direction == "buildout",
-          "phaseout",
-          "buildout"
-        ),
-        total_deviation = 0,
-        alignment_metric = 0
-      )
-
-    aggregate_exposure_company <- aggregate_exposure_company %>%
-      bind_rows(opposite_direction) %>%
-      dplyr::select(-"n_directions")
-  }
-
   sector_aggregate_exposure_loanbook_alignment <- aggregate_exposure_company %>%
     dplyr::summarise(
-      exposure_weighted_net_alignment = stats::weighted.mean(.data$alignment_metric, w = .data$exposure_weight, na.rm = TRUE),
+      exposure_weighted_net_alignment = stats::weighted.mean(
+        x = .data$alignment_metric,
+        w = .data$exposure_weight,
+        na.rm = TRUE
+      ),
       .by = dplyr::all_of(group_vars)
     )
 
-  out <- sector_aggregate_exposure_loanbook_summary %>%
+  out_aggregate <- sector_aggregate_exposure_loanbook_summary %>%
     dplyr::inner_join(
       sector_aggregate_exposure_loanbook_alignment,
       by = group_vars
     ) %>%
     dplyr::relocate(
-      c(
-        group_vars, "n_companies", "n_companies_aligned",
-        "share_companies_aligned", "exposure_weighted_net_alignment"
+      dplyr::all_of(
+        c(
+          group_vars,
+          "n_companies",
+          "n_companies_aligned",
+          "share_companies_aligned",
+          "exposure_weighted_net_alignment"
+        )
       )
     ) %>%
-    dplyr::arrange(.data$group_id, .data$scenario, .data$region, .data$sector, .data$year)
+    dplyr::arrange(
+      !!!rlang::syms(.by),
+      .data$scenario,
+      .data$region,
+      .data$sector,
+      .data$year
+    )
+
+  out <- list(
+    company = out_company,
+    aggregate = out_aggregate
+  )
 
   return(out)
 }
 
 validate_input_data_aggregate_alignment_loanbook_exposure <- function(data,
                                                                       matched,
-                                                                      group_vars) {
+                                                                      group_vars,
+                                                                      .by = NULL) {
```

Comment:
Does this function need both `group_vars` and `.by` as arguments? that seems like it would be a bit confusing?